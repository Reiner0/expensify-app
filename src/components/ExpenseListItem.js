import React from 'react';

const ExpenseListItem = ({ description, amount, createdAt }) => (
  <div>
    <h3>{description}</h3>
    <p>${(amount / 100).toFixed(2)} - {createdAt}</p>
  </div>
);

export default ExpenseListItem;
