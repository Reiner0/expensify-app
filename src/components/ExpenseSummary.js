import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = props => (
  <div>
    {
      props.expensesCount > 0 &&
        <p>
          Viewing {props.expensesCount} {props.expensesCount > 1 ? 'expenses' : 'expense'} totaling {
            numeral(props.expensesTotal / 100).format('$0,0.00')
          }
        </p>
    }
  </div>
);

const mapStateToProps = state => {
  return {
    expensesTotal: expensesTotal(selectExpenses(state.expenses, state.filters)),
    expensesCount: selectExpenses(state.expenses, state.filters).length
  };
}

export default connect(mapStateToProps)(ExpenseSummary);
