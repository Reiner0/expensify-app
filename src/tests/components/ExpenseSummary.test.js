import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { filters } from '../fixtures/filters';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render ExpenseSummary with 1 expense totaling $94.34', () => {
  const wrapper = shallow(<ExpenseSummary expensesTotal={9434} expensesCount={1} />)
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with 2 expenses totaling $94.34', () => {
  const wrapper = shallow(<ExpenseSummary expensesTotal={9434} expensesCount={2} />)
  expect(wrapper).toMatchSnapshot();
});
