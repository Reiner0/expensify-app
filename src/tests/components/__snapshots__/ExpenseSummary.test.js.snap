// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense totaling $94.34 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totaling 
    $94.34
  </p>
</div>
`;

exports[`should render ExpenseSummary with 2 expenses totaling $94.34 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totaling 
    $94.34
  </p>
</div>
`;
